ckanext/canada/tests/test_make.py

```diff
@@ -0,0 +1,482 @@
+        os.environ['PORTAL_STATIC_SMB_DIRECTORY'] = self.tmp_dir
+
+
+    @classmethod
+    def teardown_method(self, method):
+        """Method is called at class level after EACH test methods of the class are called.
+        Remove any state specific to the execution of the given class methods.
+        """
+        shutil.rmtree(self.tmp_dir)
+
+
+    def _setup_ini(self, ini):
+        assert ini.startswith('/')
+        os.environ['REGISTRY_INI'] = ini
+        os.environ['PORTAL_INI'] = ini
+
+
+    def _setup_pd(self, type, nil_type=None, extra_resource_ids=[]):
+        assert type
+
+        self.action.recombinant_create(dataset_type=type, owner_org=self.org['name'])
+
+        rval = self.action.recombinant_show(dataset_type=type, owner_org=self.org['name'])
+
+        chromo = get_chromo(type)
+
+        self.action.datastore_upsert(
+            resource_id=rval['resources'][0]['id'],
+            records=[chromo['examples']['record']])
+
+        if 'published_resource_id' in chromo:
+            Resource(id=chromo['published_resource_id'])
+
+        if nil_type:
+            nil_chromo = get_chromo(nil_type)
+
+            self.action.datastore_upsert(
+                resource_id=rval['resources'][1]['id'],
+                records=[nil_chromo['examples']['record']])
+
+            if 'published_resource_id' in nil_chromo:
+                Resource(id=nil_chromo['published_resource_id'])
+
+        for _id in extra_resource_ids:
+            Resource(id=_id)
+
+
+    def test_enivonment_variables(self):
+        assert self.ckan_ini
+        self._setup_ini(self.ckan_ini)
+
+        make_process = subprocess.Popen(["make help"], shell=True, cwd=MAKE_PATH, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+        stdout, stderr = make_process.communicate()
+
+        assert 'PD_FILTER_SCRIPT_DIRECTORY is undefined' not in stdout
+        assert 'REGISTRY_PASTER_COMMAND is undefined' not in stdout
+        assert 'REGISTRY_PYTHON_COMMAND is undefined' not in stdout
+        assert 'REGISTRY_INI is undefined' not in stdout
+        assert 'PORTAL_INI is undefined' not in stdout
+        assert 'REGISTRY_CKANAPI_COMMAND is undefined' not in stdout
+        assert 'OGC_SEARCH_COMMAND is undefined' not in stdout
+        assert 'OC_SEARCH_COMMAND is undefined' not in stdout
+        assert 'PD_BACKUP_DIRECTORY is undefined' not in stdout
+        assert 'REGISTRY_STATIC_SMB_DIRECTORY is undefined' not in stdout
+        assert 'PORTAL_STATIC_SMB_DIRECTORY is undefined' not in stdout
+
+
+    def test_make_ati(self):
+        assert self.ckan_ini
+        self._setup_ini(self.ckan_ini)
+
+        self._setup_pd(type='ati', nil_type='ati-nil')
+
+        make_process = subprocess.Popen(["make upload-ati"], shell=True, cwd=MAKE_PATH, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+        stdout, stderr = make_process.communicate()
+
+        assert "Usage:" not in stdout
+        assert "upload-ati] Error" not in stdout
```

Comment:
instead of looking for a specific string, check the error code on the subprocess